src/Scripts/DefaultVirtualScrips.spec.ts

```diff
@@ -0,0 +1,276 @@
+import 'reflect-metadata';
+import { DefaultVirtualScripts } from './DefaultVirtualScripts';
+import { VirtualScript, VirtualScriptType } from './VirtualScript';
+import { expect } from 'chai';
+
+const isValidIterable = (
+  instance: unknown
+): instance is IterableIterator<unknown> => {
+  if (instance === null || instance === undefined) {
+    return false;
+  }
+
+  return typeof instance[Symbol.iterator] === 'function';
+};
+
+describe('DefaultVirtualScripts', () => {
+  let virtualScripts: DefaultVirtualScripts;
+
+  beforeEach(() => {
+    virtualScripts = new DefaultVirtualScripts();
+  });
+
+  it('should be iterable', () => {
+    expect(isValidIterable(virtualScripts)).to.equal(true);
+  });
+
+  describe('clear', () => {
+    it('should remove all scripts when type is not passed', () => {
+      // arrange
+      virtualScripts.set('first', VirtualScriptType.LOCAL, 'let a = 1;');
+      virtualScripts.set('second', VirtualScriptType.REMOTE, 'let a = 2;');
+      // act
+      virtualScripts.clear();
+      // assert
+      expect(virtualScripts).to.deep.include({
+        store: new Map()
+      });
```

Comment:
It does work, checked with failing case and was surprised too. I agree with your reasoning. In the first revision I was checking this using `size` getter and there was a comment from @UncleVic that we should compare instance to object if something is done on the instance and not comparing properties. `Array.from(...)` approach looks like will have the same problem as checking `size` because it'll use `Symbol.iterator` and then comparing length or deep equal empty array.
So, let's clarify it finally @UncleVic @derevnjuk

Reply:
Well, object comparison is fine to my mind until and unless we check for private properties (which leaves no chance to say refactor this code leave the tests intact). I see no problem in using deep equality check with an array instance that we've made out of an iterator.